app/controllers/students_controller.rb

```diff
@@ -1,31 +1,37 @@
 class StudentsController < ApplicationController
   def index
+    create_callback_hash
     @students = Student.all.order(id: :asc)
   end
 
   def show
+    create_callback_hash
     @student = Student.find(params[:id])
   end
 
   def new
+    create_callback_hash
     @student = Student.new
   end
 
   def create
+    create_callback_hash
     @student = Student.new(student_params)
 
     if @student.save
-      redirect_to students_path
+      redirect_to student_path(@student.id)
```

Comment:
why @student.id is passed in path

Reply:
because i want to redirect a user to the show methods after creating new one.
previously it was redirected to the index page but i made this change by your suggestion.

Reply:
ok